src/cephadm/cephadm

```diff
@@ -6457,6 +6457,12 @@ def command_rm_daemon(ctx):
 
     call(ctx, ['systemctl', 'stop', unit_name],
          verbosity=CallVerbosity.DEBUG)
+
+    # remove the mon from cluster after we stop the mon.
+    if daemon_type == 'mon':
+        call(ctx, ['ceph', 'mon', 'rm', daemon_id],
```

Comment:
Can I get some clarification on what the exact order of operations we need is? We have a few things going on

- Stopping the mon (right above this)
- removing the mon from the quorum (this command here)
- disabling and fully removing the mon (next few lines after this)

What are the requirements in terms of ordering here? If it's just that we need to stop the mon before removing it from the quorum, you could consider just moving the `mon rm` command you removed from pre-remove into post-remove which will run after rm_daemon (thus making it the last of the 3 operations to happen). If it is necessary to have this exact order where we stop the mon, then remove from quorum, then disable and fully remove the mon, you could leave it here. However, this would need to be done as a shell command (shell command basically starts up a container to run the command in) rather than just calling `call` (see https://github.com/ceph/ceph/blob/master/src/cephadm/cephadm#L3640) as there is no guarantee that the ceph packages are installed directly on the host (which is where `call` will run). This could be a bit tricky to setup correctly, which is why I want to verify that we NEED this to be done in the middle here and not just after the the mon is stopped.